CP-310169: Avoid defining unused machines to save memory

Keep the latest 10.1 machine to both show the latest available and avoid
re-defining the old machines as the 'latest'.

Removing the definitions for the earlier machines saves a large maount of
memory allocations at run time which are duplicated between QEMU processes.

Signed-off-by: Gerald Elder-Vass <gerald.elder-vass@citrix.com>
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 221edec1daa8..0794fb87849b 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -525,7 +525,7 @@ static void pc_i440fx_machine_10_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_10_0, pc_compat_10_0_len);
 }
 
-DEFINE_I440FX_MACHINE(10, 0);
+//DEFINE_I440FX_MACHINE(10, 0);
 
 static void pc_i440fx_machine_9_2_options(MachineClass *m)
 {
@@ -534,7 +534,7 @@ static void pc_i440fx_machine_9_2_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_9_2, pc_compat_9_2_len);
 }
 
-DEFINE_I440FX_MACHINE(9, 2);
+//DEFINE_I440FX_MACHINE(9, 2);
 
 static void pc_i440fx_machine_9_1_options(MachineClass *m)
 {
@@ -543,7 +543,7 @@ static void pc_i440fx_machine_9_1_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_9_1, pc_compat_9_1_len);
 }
 
-DEFINE_I440FX_MACHINE(9, 1);
+//DEFINE_I440FX_MACHINE(9, 1);
 
 static void pc_i440fx_machine_9_0_options(MachineClass *m)
 {
@@ -557,7 +557,7 @@ static void pc_i440fx_machine_9_0_options(MachineClass *m)
     pcmc->isa_bios_alias = false;
 }
 
-DEFINE_I440FX_MACHINE(9, 0);
+//DEFINE_I440FX_MACHINE(9, 0);
 
 static void pc_i440fx_machine_8_2_options(MachineClass *m)
 {
@@ -571,7 +571,7 @@ static void pc_i440fx_machine_8_2_options(MachineClass *m)
     pcmc->default_smbios_ep_type = SMBIOS_ENTRY_POINT_TYPE_64;
 }
 
-DEFINE_I440FX_MACHINE(8, 2);
+//DEFINE_I440FX_MACHINE(8, 2);
 
 static void pc_i440fx_machine_8_1_options(MachineClass *m)
 {
@@ -584,7 +584,7 @@ static void pc_i440fx_machine_8_1_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_8_1, pc_compat_8_1_len);
 }
 
-DEFINE_I440FX_MACHINE(8, 1);
+//DEFINE_I440FX_MACHINE(8, 1);
 
 static void pc_i440fx_machine_8_0_options(MachineClass *m)
 {
@@ -598,7 +598,7 @@ static void pc_i440fx_machine_8_0_options(MachineClass *m)
     pcmc->default_smbios_ep_type = SMBIOS_ENTRY_POINT_TYPE_32;
 }
 
-DEFINE_I440FX_MACHINE(8, 0);
+//DEFINE_I440FX_MACHINE(8, 0);
 
 static void pc_i440fx_machine_7_2_options(MachineClass *m)
 {
@@ -607,7 +607,7 @@ static void pc_i440fx_machine_7_2_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_7_2, pc_compat_7_2_len);
 }
 
-DEFINE_I440FX_MACHINE(7, 2)
+//DEFINE_I440FX_MACHINE(7, 2)
 
 static void pc_i440fx_machine_7_1_options(MachineClass *m)
 {
@@ -616,7 +616,7 @@ static void pc_i440fx_machine_7_1_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_7_1, pc_compat_7_1_len);
 }
 
-DEFINE_I440FX_MACHINE(7, 1);
+//DEFINE_I440FX_MACHINE(7, 1);
 
 static void pc_i440fx_machine_7_0_options(MachineClass *m)
 {
@@ -627,7 +627,7 @@ static void pc_i440fx_machine_7_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_7_0, pc_compat_7_0_len);
 }
 
-DEFINE_I440FX_MACHINE(7, 0);
+//DEFINE_I440FX_MACHINE(7, 0);
 
 static void pc_i440fx_machine_6_2_options(MachineClass *m)
 {
@@ -636,7 +636,7 @@ static void pc_i440fx_machine_6_2_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_6_2, pc_compat_6_2_len);
 }
 
-DEFINE_I440FX_MACHINE(6, 2);
+//DEFINE_I440FX_MACHINE(6, 2);
 
 static void pc_i440fx_machine_6_1_options(MachineClass *m)
 {
@@ -646,7 +646,7 @@ static void pc_i440fx_machine_6_1_options(MachineClass *m)
     m->smp_props.prefer_sockets = true;
 }
 
-DEFINE_I440FX_MACHINE(6, 1);
+//DEFINE_I440FX_MACHINE(6, 1);
 
 static void pc_i440fx_machine_6_0_options(MachineClass *m)
 {
@@ -655,7 +655,7 @@ static void pc_i440fx_machine_6_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_6_0, pc_compat_6_0_len);
 }
 
-DEFINE_I440FX_MACHINE(6, 0);
+//DEFINE_I440FX_MACHINE(6, 0);
 
 static void pc_i440fx_machine_5_2_options(MachineClass *m)
 {
@@ -664,7 +664,7 @@ static void pc_i440fx_machine_5_2_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_5_2, pc_compat_5_2_len);
 }
 
-DEFINE_I440FX_MACHINE(5, 2);
+//DEFINE_I440FX_MACHINE(5, 2);
 
 static void pc_i440fx_machine_5_1_options(MachineClass *m)
 {
@@ -677,7 +677,7 @@ static void pc_i440fx_machine_5_1_options(MachineClass *m)
     pcmc->pci_root_uid = 1;
 }
 
-DEFINE_I440FX_MACHINE(5, 1);
+//DEFINE_I440FX_MACHINE(5, 1);
 
 static void pc_i440fx_machine_5_0_options(MachineClass *m)
 {
@@ -688,7 +688,7 @@ static void pc_i440fx_machine_5_0_options(MachineClass *m)
     m->auto_enable_numa_with_memdev = false;
 }
 
-DEFINE_I440FX_MACHINE(5, 0);
+//DEFINE_I440FX_MACHINE(5, 0);
 
 static void pc_i440fx_machine_4_2_options(MachineClass *m)
 {
@@ -697,7 +697,7 @@ static void pc_i440fx_machine_4_2_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_4_2, pc_compat_4_2_len);
 }
 
-DEFINE_I440FX_MACHINE(4, 2);
+//DEFINE_I440FX_MACHINE(4, 2);
 
 static void pc_i440fx_machine_4_1_options(MachineClass *m)
 {
@@ -706,7 +706,7 @@ static void pc_i440fx_machine_4_1_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_4_1, pc_compat_4_1_len);
 }
 
-DEFINE_I440FX_MACHINE(4, 1);
+//DEFINE_I440FX_MACHINE(4, 1);
 
 static void pc_i440fx_machine_4_0_options(MachineClass *m)
 {
@@ -717,7 +717,7 @@ static void pc_i440fx_machine_4_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_4_0, pc_compat_4_0_len);
 }
 
-DEFINE_I440FX_MACHINE(4, 0);
+//DEFINE_I440FX_MACHINE(4, 0);
 
 static void pc_i440fx_machine_3_1_options(MachineClass *m)
 {
@@ -730,7 +730,7 @@ static void pc_i440fx_machine_3_1_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_3_1, pc_compat_3_1_len);
 }
 
-DEFINE_I440FX_MACHINE(3, 1);
+//DEFINE_I440FX_MACHINE(3, 1);
 
 static void pc_i440fx_machine_3_0_options(MachineClass *m)
 {
@@ -739,7 +739,7 @@ static void pc_i440fx_machine_3_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_3_0, pc_compat_3_0_len);
 }
 
-DEFINE_I440FX_MACHINE(3, 0);
+//DEFINE_I440FX_MACHINE(3, 0);
 
 static void pc_i440fx_machine_2_12_options(MachineClass *m)
 {
@@ -748,7 +748,7 @@ static void pc_i440fx_machine_2_12_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_12, pc_compat_2_12_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 12);
+//DEFINE_I440FX_MACHINE(2, 12);
 
 static void pc_i440fx_machine_2_11_options(MachineClass *m)
 {
@@ -757,7 +757,7 @@ static void pc_i440fx_machine_2_11_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_11, pc_compat_2_11_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 11);
+//DEFINE_I440FX_MACHINE(2, 11);
 
 static void pc_i440fx_machine_2_10_options(MachineClass *m)
 {
@@ -776,7 +776,7 @@ static void pc_i440fx_machine_2_9_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_9, pc_compat_2_9_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 9);
+//DEFINE_I440FX_MACHINE(2, 9);
 
 static void pc_i440fx_machine_2_8_options(MachineClass *m)
 {
@@ -785,7 +785,7 @@ static void pc_i440fx_machine_2_8_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_8, pc_compat_2_8_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 8);
+//DEFINE_I440FX_MACHINE(2, 8);
 
 static void pc_i440fx_machine_2_7_options(MachineClass *m)
 {
@@ -794,7 +794,7 @@ static void pc_i440fx_machine_2_7_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_7, pc_compat_2_7_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 7);
+//DEFINE_I440FX_MACHINE(2, 7);
 
 static void pc_i440fx_machine_2_6_options(MachineClass *m)
 {
@@ -808,7 +808,7 @@ static void pc_i440fx_machine_2_6_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_compat_2_6, pc_compat_2_6_len);
 }
 
-DEFINE_I440FX_MACHINE(2, 6);
+//DEFINE_I440FX_MACHINE(2, 6);
 
 
 /* Looking for a pc_compat_2_4() function? It doesn't exist.
